src/auth/userSession.ts

```diff
@@ -205,20 +205,20 @@ export class UserSession {
 
   /**
    * Check if there is a authentication request that hasn't been handled.
-   * 
+   *
    * @returns{Boolean} `true` if there is a pending sign in, otherwise `false`
    */
-  isSignInPending() {
+  isSignInPending(): boolean {
     return authApp.isSignInPending()
   }
 
   /**
    * Check if a user is currently signed in.
-   * 
+   *
    * @returns {Boolean} `true` if the user is signed in, `false` if not.
    */
-  isUserSignedIn() {
-    return !!this.store.getSessionData().userData
+  isUserSignedIn(): Promise<boolean> {
```

Comment:
This sync -> async API change will probably break every existing app using Blockstack.js. Even with a major semver bump, I anticipate a lot of frustration. 

Should this use the `@deprecated` tag and console warning, and continue to work for synchronous SessionStorage implementations? A new async-only implementation could throw a "not supported" Error. 

Then something like `isUserSignedInAsync`, or function overloading e.g. `isUserSignedIn({async: true}): Promise<bool>`. 

Not sure what to do here..

Reply:
Yeah, there's no way around it, we have to make this async. Because the underlying `SessionDataStore` class is async, this function must be async. There's no way to take a `Promise` and make it synchronous (ie. block the thread and wait for the result).

I don't know how other people structure their apps, but in my case it wasn't that difficult to make this async. So I'm not sure how much frustration there would be. You probably have a better view on this though.

I can just use my own fork for the time being. But I assume you'll want to merge this at some point?

Reply:
I think the best approach here is to keep all of these functions sync by default, and require a `opts: {async: true}` param to take the async path and return a Promise. Typescript function overloading should be helpful here. 

When a backing SessionStorage implementation does not support async, it should throw a useful Error stating that the `{async: true}` param should be specified. 

This gives us an easy path in the future to start deprecating the sync version, then eventually remove it once we feel enough devs have updated.